hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2328,8 +2329,8 @@ func reconcileCAPIManagerRoleBinding(binding *rbacv1.RoleBinding, role *rbacv1.R
 	return nil
 }
 
-func reconcileCAPIProviderRole(role *rbacv1.Role) error {
-	role.Rules = []rbacv1.PolicyRule{
+func reconcileCAPIProviderRole(role *rbacv1.Role, p platform.Platform) error {
```

Comment:
consider p to be a pointer to avoid copying by value?

Reply:
Changed the assigned structures to be pointers

Reply:
I don't see the change, I'm still seeing you pass p platform.Platform) here. That's ok, this is not a blocker to me anyways.